prompts/templates.go

```diff
@@ -0,0 +1,8 @@
+package prompts
+
+type TemplateFormat string
+
+const (
+	TemplateFormatFString TemplateFormat = "f-string"
+	TemplateFormatJinja2  TemplateFormat = "jinja2"
```

Comment:
Should we try to support this or go to go templates?

Reply:
f-string should be simple enough to implement, kind of doubtful about jinja2 -> we can always layer that on -- let's get this PR ready and iterate!